src/cypher/execution_plan/ops/op_var_len_expand.cpp

```diff
@@ -14,6 +14,454 @@
+                    relp_->path_.GetNthEdgeWithTid(relp_->path_.Length() - 1)));
+            }
+
+            stack.pop_back();
+            if (currentLevel >= min_hop_) {
+                neighbor_->PushVid(currentNodeId);
+
+                if (relp_->path_.Length() != 0) {
+                    needPop = true;
+                }
+
+                return true;
+            }
+            if (relp_->path_.Length() != 0) {
+                relp_->path_.PopBack();
+            }
+        }
+    }
+    return false;
+}
+
+bool VarLenExpand::Next(RTContext *ctx) {
+    // check label filter
+    do {
+        if (!NextWithFilter(ctx)) return false;
+    } while (!neighbor_->Label().empty() && neighbor_->IsValidAfterMaterialize(ctx) &&
+             neighbor_->ItRef()->GetLabel() != neighbor_->Label());
+    return true;
+}
+
+VarLenExpand::VarLenExpand(PatternGraph *pattern_graph, Node *start, Node *neighbor,
+                           Relationship *relp)
+    : OpBase(OpType::VAR_LEN_EXPAND, "Variable Length Expand"),
+      pattern_graph_(pattern_graph),
+      start_(start),
+      neighbor_(neighbor),
+      relp_(relp),
+      min_hop_(relp->MinHop()),
+      max_hop_(relp->MaxHop()) {
+    modifies.emplace_back(neighbor_->Alias());
+    modifies.emplace_back(relp_->Alias());
+    auto &sym_tab = pattern_graph->symbol_table;
+    auto sit = sym_tab.symbols.find(start_->Alias());
+    auto dit = sym_tab.symbols.find(neighbor_->Alias());
+    auto rit = sym_tab.symbols.find(relp_->Alias());
+    CYPHER_THROW_ASSERT(sit != sym_tab.symbols.end() && dit != sym_tab.symbols.end() &&
+                        rit != sym_tab.symbols.end());
+    expand_direction_ = relp_->Undirected()            ? BIDIRECTIONAL
+                        : relp_->Src() == start_->ID() ? FORWARD
+                                                       : REVERSED;
+    start_rec_idx_ = sit->second.id;
+    nbr_rec_idx_ = dit->second.id;
+    relp_rec_idx_ = rit->second.id;
+}
+
+void VarLenExpand::addPredicate(std::unique_ptr<Predicate> p) {
+    predicates.push_back(std::move(p));
+}
+
+void VarLenExpand::PushFilter(std::shared_ptr<lgraph::Filter> filter) {
+    if (filter) {
+        if (filter->Type() == lgraph::Filter::RANGE_FILTER) {
+            std::shared_ptr<lgraph::RangeFilter> tmp_filter =
+                std::static_pointer_cast<lgraph::RangeFilter>(filter);
+            if (tmp_filter->GetAeLeft().op.type == cypher::ArithOpNode::AR_OP_FUNC) {
+                std::string func_name = tmp_filter->GetAeLeft().op.func_name;
+                std::transform(func_name.begin(), func_name.end(), func_name.begin(), ::tolower);
+                if (func_name == "isasc") {
+                    auto p = std::make_unique<IsAscPredicate>();
+                    addPredicate(std::move(p));
+                } else if (func_name == "isdesc") {
+                    auto p = std::make_unique<IsDescPredicate>();
+                    addPredicate(std::move(p));
```

Comment:
The move function is redundant to the one inside `addPredicate`

Reply:
It seems that without this `std::move`, the compilation will fail.